File: kq-projection-dotnet/kq-projection/Common/Models/Events/GameDiscardedEventData.cs
using KQ_Projection.Common.Enums;
using System.Text.Json.Serialization;

namespace KQ_Projection.Common.Models.Events;

public class GameDiscardedEventData : EventData
{
    [JsonPropertyName("discardReason")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public DiscardReason discardReason {  get; set; }
}

File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/eventdata/GameDiscardedEventData.java
package com.kafkaquest.kq.common.util.models.events.eventdata;

import com.kafkaquest.kq.common.util.models.events.enums.DiscardReason;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;


@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GameDiscardedEventData extends EventData {
    private DiscardReason discardReason;
}
